src/azul/service/elasticsearch_service.py

```diff
@@ -525,21 +525,65 @@ def transform_summary(self,
             )
-            # Add a total cell count aggregate
+            # FIXME: Remove deprecated field totalCellCount
+            #        https://github.com/DataBiosphere/azul/issues/3650
+            # Add a cell suspensions cell count aggregate
             es_search.aggs.metric('totalCellCount',
                                   'sum',
                                   field='contents.cell_suspensions.total_estimated_cells_')
+            # Add a cell suspensions cell count aggregate from projects
+            # with a project level estimated cell count, and another from those
+            # projects without a project level estimated cell count.
+            field = 'contents.projects.estimated_cell_count'
+            field_type = self.field_type(catalog, tuple(field.split('.')))
+            null_value = field_type.to_index(None)
+            es_search.aggs.bucket(
+                'projectCellCount',
+                'filters',
+                filters={
+                    # Since all projects will have this field we can filter
+                    # for those with an empty value (zero or null).
+                    'hasNone': Q('terms', **{field: [0, null_value]})
+                },
+                other_bucket_key='hasSome',
+            ).metric(
+                'totalCellCount',
+                'sum',
+                field='contents.cell_suspensions.total_estimated_cells_'
+            )
         elif entity_type == 'samples':
             # Add an organ aggregate to the Elasticsearch request
             es_search.aggs.bucket('organTypes',
                                   'terms',
                                   field='contents.samples.effective_organ.keyword',
                                   size=config.terms_aggregation_size)
         elif entity_type == 'projects':
+            # FIXME: Remove deprecated field projectEstimatedCellCount
+            #        https://github.com/DataBiosphere/azul/issues/3650
             # Add a project cell count aggregate
             es_search.aggs.metric('projectEstimatedCellCount',
                                   'sum',
                                   field='contents.projects.estimated_cell_count_')
+            # Add a project cell count aggregate from the projects that have
+            # cell suspension cell counts, and another from the projects without
+            # any cell suspension cell counts.
+            field = 'contents.cell_suspensions.total_estimated_cells'
+            field_type = self.field_type(catalog, tuple(field.split('.')))
+            null_value = field_type.to_index(None)
+            es_search.aggs.bucket(
+                'cellSuspensionCellCount',
+                'filters',
+                filters={
+                    # Since not all projects will have a cell suspension we
+                    # filter for those with the field and a non-empty value.
+                    'hasSome': Q('range', **{field: {'gt': 0, 'lt': null_value}})
```

Comment:
This is depending on what the null value actually is, which is a detail I'd rather hide in the field type. Either eliminate the ordering dependency by rewriting the filter requirement or add an abstraction to the field type. I hope we can do the former.